account_multicompany_easy_creation/wizards/multicompany_easy_creation.py

```diff
@@ -157,8 +157,19 @@ def create_company(self):
         self.new_company_id = self.env["res.company"].create(
             {"name": self.name, "user_ids": [(6, 0, self.user_ids.ids)]}
         )
-        self.new_company_id.chart_template_id = self.chart_template_id
-        self.new_company_id.chart_template_id.try_loading_for_current_company()
+        self.env.user.write(
+            {
+                "company_ids": [(4, self.new_company_id.id)],
+                "company_id": self.new_company_id.id,
+            }
+        )
+        ctx = dict(self.env.context)
+        allowed_company_ids = ctx.get("allowed_company_ids") or []
+        allowed_company_ids.append(self.new_company_id.id)
+        ctx.update(allowed_company_ids=allowed_company_ids)
+        self.with_context(
+            context=ctx
```

Comment:
OK, then let's favor the readability:
```suggestion
        self.with_context(allowed_company_ids=(
                self.env.context.get("allowed_company_ids", []) + 
                self.new_company_id.ids
        )
```

Reply:
Thanks, but although this change works in the tests it does not work in UI, perhaps because the context already has `allowed_company_ids` defined and that is why the entire context needs to be redefined.

Reply:
There should be another thing then, as `with_context` replace the corresponding key, no matter if it exists previously or not.

Reply:
https://github.com/odoo/odoo/blob/6fb05387f0b49bd4b7fd36ae3677c6679b46c639/odoo/models.py#L5192

Reply:
The code you mention is not wasting my proposal. It's designed for always including `allowed_company_ids` in the context, although you replace the whole dictionary and remove it, but it allows to replace it by another. Look this shell extract:

![Selección_001](https://user-images.githubusercontent.com/7165771/135398171-0de16f38-6245-4f64-896e-2cf3abb6ddf9.png)

Reply:
Ok, I have changed it according to your suggestion even though it doesn't work in UI.

Reply:
This is not what I want. What I want is that you find the real root of the problem, which is not what you were saying it is.

Reply:
Sorry for the confusion, it is already fixed. It is also necessary to specify `allowed_company_ids` on the company that we have previously created.